x-pack/plugins/secops/public/pages/home/index.tsx

```diff
@@ -26,49 +29,60 @@ import {
   SubHeaderDatePicker,
 } from '../../components/page';
 import { DatePicker } from '../../components/page/date_picker';
-import { Flyout } from '../../components/page/flyout';
 import { Footer } from '../../components/page/footer';
 import { Navigation } from '../../components/page/navigation';
 import { StatefulTimeline } from '../../components/timeline';
+import { defaultWidth } from '../../components/timeline/body';
 import { headers } from '../../components/timeline/body/column_headers/headers';
 import { NotFoundPage } from '../404';
 import { Hosts } from '../hosts';
 import { Network } from '../network';
 import { Overview } from '../overview';
 
-const maxTimelineWidth = 1125;
+interface Props {
+  dispatch: Dispatch;
+}
 
-export const HomePage = pure(() => (
-  <PageContainer data-test-subj="pageContainer">
-    <Flyout>
-      <StatefulTimeline id="timeline" headers={headers} width={maxTimelineWidth} />
-    </Flyout>
-    <PageHeader data-test-subj="pageHeader">
-      <Navigation data-test-subj="navigation" />
-    </PageHeader>
-    <PageContent data-test-subj="pageContent">
-      <SubHeader data-test-subj="subHeader">
-        <SubHeaderDatePicker data-test-subj="datePickerContainer">
-          <DatePicker />
-        </SubHeaderDatePicker>
-        <EuiHorizontalRule margin="none" />
-      </SubHeader>
-      <Pane data-test-subj="pane">
-        <PaneHeader data-test-subj="paneHeader">
-          <EuiSearchBar onChange={noop} />
-        </PaneHeader>
-        <PaneScrollContainer data-test-subj="pane1ScrollContainer">
-          <Switch>
-            <Redirect from="/" exact={true} to="/overview" />
-            <Route path="/overview" component={Overview} />
-            <Route path="/hosts" component={Hosts} />
-            <Route path="/network" component={Network} />
-            <Route path="/link-to" component={LinkToPage} />
-            <Route component={NotFoundPage} />
-          </Switch>
-        </PaneScrollContainer>
-      </Pane>
-    </PageContent>
-    <Footer />
-  </PageContainer>
-));
+class HomePageComponent extends React.PureComponent<Props> {
```

Comment:
I think this is still not `pure(` as you want it? I marked it as a thumbs up but it's still doing an `extends`